### PR TITLE
Fix and improve the documentation of dprintf.

### PR DESCRIPTION
**Pull request by Drup**
After remarks by @yallop and @pmetzger .

### CODE REVIEW

File: stdlib/format.mli [hunk 1]
```diff
@@ -1071,10 +1071,20 @@ val asprintf : ('a, formatter, unit, string) format4 -> 'a
 val dprintf :
   ('a, formatter, unit, formatter -> unit) format4 -> 'a
 (** Same as {!fprintf}, except the formatter is the last argument.
-  All printing actions are delayed until the formatter is provided.
-  In particular, [printff "..." a b c] is a
-  function of type [formatter -> unit] which can be given to
-  a format specifier [%t].
+  In particular, [dprintf "..." a b c] is a function of type
+  [formatter -> unit] which can be given to a format specifier [%t].
```
Comment (pmetzger) [inline stdlib/format.mli]: I'd remove "In particular, ", I think it's probably redundant.

File: stdlib/format.mli [hunk 2]
```diff
@@ -1071,10 +1071,20 @@ val asprintf : ('a, formatter, unit, string) format4 -> 'a
...
+
+  This can be used as a replacement of {!asprintf} to delay formatting
```
Comment (pmetzger) [inline stdlib/format.mli]: "a replacement for" rather than "of" I think.

File: stdlib/format.mli [hunk 3]
```diff
@@ -1071,10 +1071,20 @@ val asprintf : ('a, formatter, unit, string) format4 -> 'a
...
+  decisions.  By using the string returned by {!asprintf} in a
+  formatting context, formatting decisions are taken in isolation and
+  the final string is created prematurely.  {!dprintf} allows to delay
+  such formatting decisions until the final formatting context is
+  known. For example:
```
Comment (pmetzger) [inline stdlib/format.mli]: Perhaps: "Using the string returned by {!asprintf} in a
  formatting context forces formatting decisions to be taken in isolation, and
  the final string may be created prematurely.  {!dprintf} allows delay of
  formatting decisions until the final formatting context is
  known."

**Review by pmetzger — CHANGES_REQUESTED** [review]
Some suggested edits.

**Comment by gasche** [comment]
What about NEXT_VERSION as well, could you use 4.08? (It occurs twice in the file.)

**Comment by Drup** [comment]
@gasche @pmetzger Done.

**Review by gasche — APPROVED** [review]

**Pull request merged**